sds_data_manager/lambda_code/IAlirtCode/ialirt_ingest.py

```diff
@@ -100,6 +74,131 @@ def query_filenames(bucket: str, region: str, now: datetime):
     return filenames
 
 
+def parse_packets(filenames: list, bucket: str, download_dir: Path, apid=478):
+    """Get packets into datasets and combine.
+
+    This function is an event handler for s3 ingest bucket.
+    It is also used to ingest data to the DynamoDB table.
+
+    Parameters
+    ----------
+    filenames : list
+        List of file paths.
+    bucket : str
+        The name of the S3 bucket.
+    download_dir : Path
+        The directory where the file will be downloaded.
+    apid : int
+        The apid of the packet to be processed.
+
+    Returns
+    -------
+    combined : xr.Dataset
+        Combined dataset.
+    """
+    s3 = boto3.client("s3")
+    xtce_ialirt_path = (
+        imap_module_directory / "ialirt" / "packet_definitions" / "ialirt.xml"
+    )
+    datasets = []
+
+    for filename in filenames:
+        local_path = download_dir / Path(filename).name
+        s3.download_file(bucket, filename, str(local_path))
+        xarray_data = packet_file_to_datasets(
+            local_path, xtce_ialirt_path, use_derived_value=False
+        )[apid]
+        datasets.append(xarray_data)
+
+    combined = xr.concat(datasets, dim="epoch")
+    # Drop duplicate epochs. This could happen if there are duplicate packets.
+    _, unique_idx = np.unique(combined["epoch"], return_index=True)
+    combined = combined.isel(epoch=sorted(unique_idx))
+
+    return combined
+
+
+def process_algorithms(combined: xr.Dataset, algorithm_table):
+    """Process the algorithms and insert data, as needed.
+
+    Parameters
+    ----------
+    combined : xr.Dataset
+        L0 parsed data.
+    algorithm_table : dynamodb.Table
+        The DynamoDB table to insert or update the data.
+    """
+    processors = [
+        ("hit", process_hit),
+        ("swe", process_swe),
+    ]
+
+    for prefix, process_func in processors:
+        insert_data(process_func(combined), algorithm_table, prefix)
+
+
+def insert_data(data: list[dict], algorithm_table, product_prefix: str):
+    """Insert or update database row, depending on content of item.
+
+    Parameters
+    ----------
+    data : list[dict]
+        Data product produced from processing respectively instrument.
+    algorithm_table : dynamodb.Table
+        The DynamoDB table to insert or update the data.
+    product_prefix : str
+        The prefix for the product name.
```

Comment:
can you use `instrument` instead of `prefix`?
```suggestion
    for instrument, process_func in processors:
        insert_data(process_func(combined), algorithm_table, instrument)


def insert_data(data: list[dict], algorithm_table, instrument: str):
    """Insert or update database row, depending on content of item.

    Parameters
    ----------
    data : list[dict]
        Data product produced from processing respectively instrument.
    algorithm_table : dynamodb.Table
        The DynamoDB table to insert or update the data.
    instrument : str
        The instrument of the product.
```